dm/dm.go

```diff
@@ -96,18 +111,39 @@ func main() {
 		usage()
 	}
 
+	command := args[0]
 	if *stdin {
-		fmt.Printf("reading from stdin is not yet implemented")
-		os.Exit(0)
+		if len(args) != 2 {
+			usage()
+		}
+
+		cmd := exec.Command("tar", "xf", "-")
```

Comment:
It was actually quite easy to fork the process. Good catch on the library,
though. It works well as written, so changing it doesn't seem like a high
priority. However, probably worth revisiting as an optimization. That said,
it would make the executable significantly larger.

On Friday, November 27, 2015, vaikas-google <notifications@github.com>
wrote:

> In dm/dm.go
> <https://github.com/kubernetes/deployment-manager/pull/131#discussion_r46074896>
> :
>
> >  	if *stdin {
> > -		fmt.Printf("reading from stdin is not yet implemented")
> > -		os.Exit(0)
> > +		if len(args) != 2 {
> > +			usage()
> > +		}
> > +
> > +		cmd := exec.Command("tar", "xf", "-")
>
> Instead of the added complexity of forking a new process, why not use a
> library so we don't have to make the code more complex for error handling?
> https://golang.org/pkg/archive/tar/
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/deployment-manager/pull/131/files#r46074896>
> .
>


-- 
Jack Greenfield | Google